src/Warrior.java

```diff
@@ -0,0 +1,42 @@
+import java.util.List;
+
+/**
+ * Created by connorparrott123 on 12/6/2016.
+ */
+public class Warrior extends Character {
+
+    public Warrior(String name) {
+        super(name);
+     }
+
+    public void attack(Character other) {
+        int hitNumber = (int) (Math.random() * 4);
+        other.health -= 15 * hitNumber;
+        if (hitNumber == 0) {
+            System.out.println("You missed!");
+        } else if (hitNumber > 1) {
+            System.out.println("You hit " + hitNumber + " times!");
+        }
+    }
+
+    public void describeSelf() {
+        System.out.println("6ft tall, 230lbs, and black hair");
+    }
+
+    public String getPrimaryWeapon() {
+        return "Sword";
+    }
+
+    public int pickTarget(List<Character> potentialTargets) {
+        int highestHealth = 0;
+        int characterIndex = -1;
+        for (int i = 0; i < potentialTargets.size(); i++) {
```

Comment:
In this loop you are calling the function potentialTargets.get(i) multiple times when you could just call it once.
You should change this to call that function once within the loop and store the result into a variable, then use the variable instead.